MIDAS/lib/TCAL9539/TCAL9539.cpp

```diff
@@ -0,0 +1,127 @@
+#include"TCAL9539"
+#include<WIRE.h>
+
+#define WIRE Wire
+constexpr uint8_t GPIO0_ADDRESS = 0x74;
+constexpr uint8_t GPIO1_ADDRESS = 0x75;
+constexpr uint8_t GPIO2_ADDRESS = 0x77;
+
+constexpr uint8_t REG_INPUT0 = 0x0;
+constexpr uint8_t REG_INPUT1 = 0x1;
+constexpr uint8_t REG_OUTPUT0 = 0x2;
+constexpr uint8_t REG_OUTPUT1 = 0x3;
+constexpr uint8_t POLARITY_INVERSION0 = 0x4;
+constexpr uint8_t POLARITY_INVERSION1 = 0x5;
+constexpr uint8_t REG_CONFIG0 = 0x6;
+constexpr uint8_t REG_CONFIG1 = 0x7;
+
+struct PinAddress {
+    uint8_t gpio_addres; //i2c address of expander
+    uint8_t port_idx; //0 = bottom 8 bits, 1 = top 8 bits
+    uint8_t pin_offset; //[0,7] bit in port, 8 * port_idx + pin_offset = pin idx;
+    bool valid; //whether the address is valid
+};
+
+static PinAddress decode(GpioAddress pin){
+    uint8_t address = 0;
+    if(pin.gpio_id == 0){
+        address = GPIO0_ADDRESS;
+    } else if(pin.gpio_id == 1){
+        address = GPIO1_ADDRESS;
+    } else if(pin.gpio_id == 2){
+        address = GPIO2_ADDRESS;
+    } else {
+        Serial.println("Err bad gpio index");
+        return {.valid = false};
+    }
+    uint8_t pin_offset;
+    uint8_t port_idx;
+    if(pin.pin_id >= 0 && pin.pin_id < 8){
```

Comment:
8 seems like a magic number, maybe move to a constexpr called like MAX_PIN_COUNT or smth? idk I'm bad with naming